pkg/common/k8s.go

```diff
@@ -262,3 +266,133 @@ func EnrichProbe(probe **v1.Probe, defaultProbe *v1.Probe) {
 		(*probe).FailureThreshold = defaultProbe.FailureThreshold
 	}
 }
+
+// GetStringValueFromSecret returns the string value from the secret by the given key and true if the key exists
+func GetStringValueFromSecret(secret *v1.Secret, key string) (string, bool) {
+	if secret == nil {
+		return "", false
+	}
+	value, ok := secret.Data[key]
```

Comment:
I might be wrong here, but if the golang impl is just an openapi wrapper (which I assume it is), then the `Data` member will contain the data as base64-encoded strings. You'll need to decode it here.